src/client/api/pythonApi.ts

```diff
@@ -345,7 +349,24 @@ export class InterpreterService implements IInterpreterService {
     @captureTelemetry(Telemetry.InterpreterListingPerf)
     public getInterpreters(resource?: Uri): Promise<PythonEnvironment[]> {
         this.hookupOnDidChangeInterpreterEvent();
-        return this.apiProvider.getApi().then((api) => api.getInterpreters(resource));
+        const promise = this.apiProvider.getApi().then((api) => api.getInterpreters(resource));
+        if (isCI) {
+            promise
+                .then((items) => {
+                    const current = interpreterCacheForCI.get(resource?.toString() || '');
+                    const itemToStore = items;
+                    if (
+                        current &&
+                        (itemToStore === current || JSON.stringify(itemToStore) === JSON.stringify(current))
+                    ) {
+                        return;
+                    }
+                    interpreterCacheForCI.set(resource?.toString() || '', itemToStore);
```

Comment:
Looks like Python extension stuff doesn't get logged, this was crucial to figuring out what was going on.
I've enabled this just for CI.
We can remove this, i'm not fussed. I thought it would be useful

Reply:
I've used the logs from Python extension a lot in the past, hence feel this is useful, but NOT necessary.

Can remove if anyone things we don't need this,

Reply:
Don't see an issue with leaving it in myself.